api/client/validator/client.go

```diff
@@ -0,0 +1,93 @@
+package validator
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/api/client"
+	"github.com/prysmaticlabs/prysm/v4/validator/rpc/apimiddleware"
+)
+
+const (
+	localKeysPath    = "/eth/v1/keystores"
+	remoteKeysPath   = "/eth/v1/remotekeys"
+	feeRecipientPath = "/eth/v1/validator/{pubkey}/feerecipient"
+)
+
+// Client is a wrapper with validator API functions on the client
+type Client struct {
+	*client.Client
+}
+
+// NewClient returns a new Client that includes functions for rest calls to keymanager APIs.
+func NewClient(host string, opts ...client.ClientOpt) (*Client, error) {
+	c, err := client.NewClient(host, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{c}, nil
+}
+
+// GetListOfValidators gets the currently known validators in hex format on the validator client whether on the web3signer or the local keystores.
+func (c *Client) GetListOfValidators(ctx context.Context) ([]string, error) {
+	localBytes, err := c.Get(ctx, localKeysPath, client.WithTokenAuthorization(c.Token()))
+	if err != nil {
+		return nil, err
+	}
+	jsonlocal := &apimiddleware.ListKeystoresResponseJson{}
+	if err := json.Unmarshal(localBytes, jsonlocal); err != nil {
+		return nil, errors.Wrap(err, "failed to parse local list keystores")
+	}
+	remoteBytes, err := c.Get(ctx, remoteKeysPath, client.WithTokenAuthorization(c.Token()))
+	if err != nil {
+		if !strings.Contains(err.Error(), "Prysm Wallet is not of type Web3Signer") {
+			return nil, err
+		}
+	}
+	jsonremote := &apimiddleware.ListRemoteKeysResponseJson{}
```

Comment:
This method makes calls to multiple endpoints and merges them together into a single flat list of strings. What if the code using this method cared about the distinction between local and remote keys? It would make things more modular to provide separate methods to retrieve local and remote validator lists, then write some other piece of code to merge them together. I would also consider responding with the full response values and letting the code using this method extract pubkey strings.